span_test.go

```diff
@@ -118,6 +118,37 @@ func TestSpanParentID(t *testing.T) {
 	assert.Equal(t, model.SpanID(parentID), payloads.Spans[0].ParentID)
 }
 
+func TestSpanLink(t *testing.T) {
+	tracer := apmtest.NewRecordingTracer()
+	defer tracer.Close()
+
+	links := []apm.SpanLink{
+		{Trace: apm.TraceID{1}, Span: apm.SpanID{1}},
+		{Trace: apm.TraceID{2}, Span: apm.SpanID{2}},
+	}
+
+	tx := tracer.StartTransaction("name", "type")
+	span := tx.StartSpanOptions("name", "type", apm.SpanOptions{
+		Links: links,
+	})
+
+	span.End()
+	tx.End()
+
+	tracer.Flush(nil)
+
+	payloads := tracer.Payloads()
+	require.Len(t, payloads.Spans, 1)
+	require.Len(t, payloads.Spans[0].Links, len(links))
+
+	// Assert equality and elements order
+	for i, sl := range links {
+		l := payloads.Spans[0].Links[i]
+		assert.Equal(t, model.SpanID(sl.Span), l.SpanID)
+		assert.Equal(t, model.TraceID(sl.Trace), l.TraceID)
+	}
```

Comment:
```suggestion
	// Assert span links are identical.
	expectedLinks := []model.SpanLink{
	        {TraceID: model.TraceID{1}, SpanID: model.SpanID{1}},
        	{TraceID: model.TraceID{2}, SpanID: model.SpanID{2}},
	}
	assert.Equal(t, expectedLinks, payloads.Spans[0].Links)
```

This does repeat the values, but IMO it's a bit clearer what's being asserted.

Reply:
Agree, I like that! :+1: 

Changed along with the transaction test!